Orange/classification/naive_bayes.py

```diff
@@ -47,23 +48,81 @@ def __init__(self, log_cont_prob, class_prob, domain):
         self.class_prob = class_prob
 
     def predict_storage(self, data):
+        if type(data) is Table:  # pylint: disable=unidiomatic-typecheck
+            return self.predict(data.X)
         if isinstance(data, Instance):
-            data = [data]
-        if len(data.domain.attributes) == 0:
+            return Table(np.atleast_2d(data.x))
+
+        if not len(data) or not len(data[0]):
             probs = np.tile(self.class_prob, (len(data), 1))
         else:
             isnan = np.isnan
-            probs = np.exp(
+            zeros = np.zeros_like(self.class_prob)
+            probs = np.atleast_2d(np.exp(
                 np.log(self.class_prob) +
-                np.array([np.zeros_like(self.class_prob)
-                          if isnan(ins.x).all() else
-                          np.sum(attr_prob[:, int(attr_val)]
-                                 for attr_val, attr_prob in zip(ins, self.log_cont_prob)
-                                 if not isnan(attr_val))
-                          for ins in data]))
+                np.array([
+                    zeros if isnan(ins.x).all() else
+                    sum(attr_prob[:, int(attr_val)]
```

Comment:
This was changed because of pending deprecation warning about `np.sum` over a generator.